src/components/Api.js

```diff
@@ -0,0 +1,66 @@
+class Api {
+  constructor({ baseURL, headers }) {
+    this._baseURL = baseURL;
+    this._headers = headers;
+  }
+
+  _handleServerResponse(res) {
+    return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+  }
+
+  getCardList() {
+    return fetch(`${this._baseURL}/cards`, {
+      headers: this._headers,
+    }).then(this._handleServerResponse);
+  }
+
+  setUserInfo({ name, about }) {
+    return fetch(`${this._baseURL}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({ name, about }),
+    }).then(this._handleServerResponse);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseURL}/users/me`, {
+      headers: this._headers,
+    }).then(this._handleServerResponse);
+  }
+
+  deleteCard(id) {
+    return fetch(`${this._baseURL}/cards/${id}`, {
+      headers: this._headers,
+
+      method: "DELETE",
+    }).then((res) => {
+      // console.log(res);
+    });
+  }
```

Comment:
`.then(this._handleServerResponse);` should be used everywhere to check if the response is `ok`